--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -6367,6 +6367,15 @@
 			DMI_MATCH(DMI_PRODUCT_NAME, "Surface Laptop Studio 2"),
 		},
 	},
+	{
+		.ident = "Microsoft Surface Pro 11th Edition with Intel",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Surface Pro"),
+			DMI_MATCH(DMI_PRODUCT_SKU,
+				  "Surface_Pro_11th_Edition_With_Intel"),
+		},
+	},
 	{}
 };
 
@@ -6390,3 +6399,8 @@
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0xa73f, quirk_no_shutdown);  // Thunderbolt 4 PCI Express Root Port
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0xa73e, quirk_no_shutdown);  // Thunderbolt 4 NHI
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0xa7a0, quirk_no_shutdown);  // GPU
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0xa831, quirk_no_shutdown);
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0xa833, quirk_no_shutdown);
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0xa84e, quirk_no_shutdown);
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0xa84f, quirk_no_shutdown);
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_INTEL, 0x64a0, quirk_no_shutdown);
